engine/src/main/java/org/terasology/engine/core/modes/loadProcesses/CreateWorldEntity.java

```diff
@@ -52,64 +65,63 @@ public String getMessage() {
 
     @Override
     public boolean step() {
-        EntityManager entityManager = context.get(EntityManager.class);
-        ChunkProvider chunkProvider = context.get(ChunkProvider.class);
+        this.entityManager = context.get(EntityManager.class);
+        this.worldGenerator = context.get(WorldGenerator.class);
+        this.config = context.get(Config.class);
+        this.chunkProvider = context.get(ChunkProvider.class);
+        this.worldConfigurator = worldGenerator.getConfigurator();
 
         Iterator<EntityRef> worldEntityIterator = entityManager.getEntitiesWith(WorldComponent.class).iterator();
         if (worldEntityIterator.hasNext()) {
+
             EntityRef worldEntity = worldEntityIterator.next();
+            worldEntityIterator.forEachRemaining(w -> logger.warn("Ignored extra world {}", w));
             chunkProvider.setWorldEntity(worldEntity);
 
-            // get the world generator config from the world entity
             // replace the world generator values from the components in the world entity
-            WorldGenerator worldGenerator = context.get(WorldGenerator.class);
-            WorldConfigurator worldConfigurator = worldGenerator.getConfigurator();
-            Map<String, Component> params = worldConfigurator.getProperties();
-            for (Map.Entry<String, Component> entry : params.entrySet()) {
-                Class<? extends Component> clazz = entry.getValue().getClass();
-                Component comp = worldEntity.getComponent(clazz);
-                if (comp != null) {
-                    worldConfigurator.setProperty(entry.getKey(), comp);
+            checkNotNull(worldConfigurator).getProperties().forEach((key, currentComponent) -> {
```

Comment:
this looks like a random null check - how does `checkNotNull` behave if it is null? just give a better error message than regular NPE?

Reply:
it throws an exception if worldConfigurator is null.

Reply:
Oh, I probably tossed that in there at some point when I got a null during development. If you give `checkNotNull` a second argument with a message, _then_ it's better than a Java < 14 NPE, but I am not sure if it's really better in this single-argument form.